app/ptxboa_functions.py

```diff
@@ -26,37 +28,50 @@ def calculate_results_single(api, settings):
     return res
 
 
-def calculate_results(api, settings):
-    # calculate results for all source regions:
-    results_list = []
-    for region in settings["region_list"]:
-        settings2 = settings.copy()
-        settings2["sel_region"] = region
-        result_single = pd.DataFrame.from_dict(
-            calculate_results_single(api, settings2)
-        ).reset_index(drop=True)
-        results_list.append(result_single)
-
-    results = pd.concat(results_list, ignore_index=True)[
-        ["source", "variable", "process_class", "value"]
-    ]
-
-    # aggregate costs (without LC):
-    res_costs = results.loc[results["variable"] != "LC"].pivot_table(
-        index="source", columns="process_class", values="value", aggfunc=sum
-    )
+def calculate_results(
+    api: PtxboaAPI, settings: dict, region_list: list = None
+) -> pd.DataFrame:
+    """Calculate results for all source regions."""
```

Comment:
I extrended the docstring (and already checked with pre-commit 😉) 
```suggestion
    """Calculate results for source regions and one selected target country.

    Parameters
    ----------
    api : :class:`~ptxboa.api.PtxboaAPI`
        an instance of the api class
    settings : dict
        settings from the streamlit app. An example can be obtained with the
        return value from :func:`ptxboa_functions.create_sidebar`.
    region_list : list or None
        The regions for which the results are calculated. If None, all regions
        available in the API will be used.

    Returns
    -------
    pd.DataFrame
        same format as for :meth:`~ptxboa.api.PtxboaAPI.calculate()`
    """
```